bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java

```diff
@@ -532,4 +532,25 @@ private void resetSize(int index) {
             size = index + 1;
         }
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void attemptFreeze(Status freezeStatus) {
+        if (!FreezeUtils.isOpenForFreeze(this.freezeStatus, freezeStatus)) {
+            return;
+        }
+        this.freezeStatus = freezeStatus;
+        if (elementType == null || !(elementType.getTag() == TypeTags.INT_TAG ||
+                elementType.getTag() == TypeTags.STRING_TAG || elementType.getTag() == TypeTags.BOOLEAN_TAG ||
+                elementType.getTag() == TypeTags.FLOAT_TAG || elementType.getTag() == TypeTags.BYTE_TAG)) {
+            for (int i = 0; i < this.size; i++) {
+                Object refValue = this.getRefValue(i);
+                if (refValue instanceof RefValue) {
```

Comment:
since we are checking the element-type tag, i think checking `instanceof` is not needed

Reply:
fixed